src/LinearAlgebra/COO_Matrix.hpp

```diff
@@ -351,37 +365,37 @@ inline void COO_Matrix<ScalarT, Intdx>::axpy(ScalarT alpha, std::vector<Intdx> r
     //sort input
     this->sortSparseCOO(r, c, v);
 
-    Intdx aiter = 0;
+    Intdx a_iter = 0;
     //iterate for all current values_ in matrix
-    for (Intdx i = 0; i < static_cast<Intdx>(this->row_indexes_.size()); i++)
+    for (Intdx i = 0; i < static_cast<Intdx>(this->row_indices_.size()); i++)
     {
         //pushback values_ when they are not in current matrix
-        while(aiter < static_cast<Intdx>(r.size()) && (r[aiter] < this->row_indexes_[i] || (r[aiter] == this->row_indexes_[i] && c[aiter] < this->column_indexes_[i])))
+        while(a_iter < static_cast<Intdx>(r.size()) && (r[a_iter] < this->row_indices_[i] || (r[a_iter] == this->row_indices_[i] && c[a_iter] < this->column_indices_[i])))
         {
-            this->row_indexes_.push_back(r[aiter]);
-            this->column_indexes_.push_back(c[aiter]);
-            this->values_.push_back(alpha * v[aiter]);
+            this->row_indices_.push_back(r[a_iter]);
+            this->column_indices_.push_back(c[a_iter]);
+            this->values_.push_back(alpha * v[a_iter]);
             
-            this->checkIncreaseSize(r[aiter], c[aiter]);
-            aiter++;
+            this->checkIncreaseSize(r[a_iter], c[a_iter]);
+            a_iter++;
         }
-        if (aiter >= static_cast<Intdx>(r.size()))
+        if (a_iter >= static_cast<Intdx>(r.size()))
         {
```

Comment:
Suggest using `index_type` or `IdxT` as the template parameter name:
```suggestion
        if (a_iter >= static_cast<IdxT>(r.size()))
        {
```
Make sure changes are made throughout the code.